appcues/src/main/java/com/appcues/data/mapper/experience/ExperienceMapper.kt

```diff
@@ -138,14 +138,13 @@ internal class ExperienceMapper(
         val containerTraits = stepContainerResponse.traits.map { it to GROUP }
         val mergedTraits = containerTraits.mergeTraits(experienceTraits)
         val mappedTraits = traitsMapper.map(renderContext, mergedTraits)
-
+        val presentingTrait = mappedTraits.getExperiencePresentingTraitOrThrow()
         // this is where we will use groups to organize steps into stepContainers
         // also merge all necessary traits for each step
         StepContainer(
             id = stepContainerResponse.id,
-            steps = stepContainerResponse.children.map { step -> stepMapper.map(renderContext, step, mergedTraits) },
+            steps = stepContainerResponse.children.map { step -> stepMapper.map(renderContext, step, presentingTrait, mergedTraits) },
             actions = actionsMapper.map(renderContext, stepContainerResponse.actions),
-            presentingTrait = mappedTraits.getExperiencePresentingTraitOrThrow(),
```

Comment:
the step is what is actually being presented as i noted above, the existence of StepContainer for the state machine is irrelevant, we can skip it and only consult flatSteps when going back and forth on steps.

the only reason we do use StepContainer is when we want to check if a step belongs in the same container.

Reply:
gotcha, yeah it seems not a huge deal - logically the presentation is part of the container, but i can understand how from the code perspective it simplifies things to have access to it from the step. It would only ever really be used by the first step in the container, so it is a little unexpected that say steps 2, 3, and so on have a presenting trait, but it's an implementation detail, fair enough.